geth/transactions/txqueue_manager_test.go

```diff
@@ -181,6 +172,59 @@ func (s *TxQueueTestSuite) TestCompleteTransactionMultipleTimes() {
 	s.Equal(txCount-1, inprogressTx, "txs expected to be reported as inprogress")
 }
 
+// testTxSender implements ethereum.TransactionSender
+// for testing purposes.
+type testTxSender struct {
+	tx  *common.QueuedTx
+	s   *TxQueueTestSuite
+	acc *common.SelectedExtKey
+}
+
+// SendTransaction is redefined here to inject into
+// CompleteTransaction execution via Manager. Checks
+// transaction values.
+func (sender *testTxSender) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	s := sender.s
+	s.Equal(sender.tx.Args.Gas.ToInt(), tx.Gas())
```

Comment:
all these things should be compared by setting correct expectations for mocks

Reply:
Can you elaborate on this a bit? I need a simple example on how to apply that here.